src/cron_tasks/monitoring_metrics.py

```diff
@@ -75,19 +75,28 @@ async def run_calculate_performance_metrics_pipeline(
     After the metrics are calculated they are saved in the database
     """
 
-    if actual_df.empty:
+    cleaned_actuals_df = actual_df.dropna()
+
+    if cleaned_actuals_df.empty:
         logger.info(
             f"Can't calculate performance metrics for model {model.id} because no actuals were found!"
         )
         return
 
+    if len(cleaned_actuals_df.index) != len(actual_df.index):
+        logger.info(
+            f"Performance metrics will be calculates only for a portion of rows for model: {model.id}\
+                because actuals were not provided for all inference rows!"
+        )
+        inference_processed_df = inference_processed_df.iloc[cleaned_actuals_df.index]
+
```

Comment:
Should indices be reset in `inference_processed_df` and `cleaned_actuals_df` before calculation?

Reply:
Actually, the below functions will not have a problem since the 2 dataframes will have the same index. 
Just in order to be sure, you can use .reset_index(drop=True) if you want

Reply:
I used reset just in case! 😊